tests/pos/stdlib/collection/IterableOnce.scala

```diff
@@ -0,0 +1,1364 @@
+  * design of the collections library, which uses the following pattern:
+  * {{{
+  *   trait Seq extends Iterable with SeqOps
+  *   trait SeqOps extends IterableOps
+  *
+  *   trait IndexedSeq extends Seq with IndexedSeqOps
+  *   trait IndexedSeqOps extends SeqOps
+  * }}}
+  *
+  * The goal is to provide a minimal interface without any sequential operations. This allows
+  * third-party extension like Scala parallel collections to integrate at the level of IterableOnce
+  * without inheriting unwanted implementations.
+  *
+  * @define coll collection
+  */
+trait IterableOnce[+A] extends Any {
+  this: IterableOnce[A]^ =>
+
+  /** Iterator can be used only once */
+  def iterator: Iterator[A]^{this}
+
+  /** Returns a [[scala.collection.Stepper]] for the elements of this collection.
+    *
+    * The Stepper enables creating a Java stream to operate on the collection, see
+    * [[scala.jdk.StreamConverters]]. For collections holding primitive values, the Stepper can be
+    * used as an iterator which doesn't box the elements.
+    *
+    * The implicit [[scala.collection.StepperShape]] parameter defines the resulting Stepper type according to the
+    * element type of this collection.
+    *
+    *   - For collections of `Int`, `Short`, `Byte` or `Char`, an [[scala.collection.IntStepper]] is returned
+    *   - For collections of `Double` or `Float`, a [[scala.collection.DoubleStepper]] is returned
+    *   - For collections of `Long` a [[scala.collection.LongStepper]] is returned
+    *   - For any other element type, an [[scala.collection.AnyStepper]] is returned
+    *
+    * Note that this method is overridden in subclasses and the return type is refined to
+    * `S with EfficientSplit`, for example [[scala.collection.IndexedSeqOps.stepper]]. For Steppers marked with
+    * [[scala.collection.Stepper.EfficientSplit]], the converters in [[scala.jdk.StreamConverters]]
+    * allow creating parallel streams, whereas bare Steppers can be converted only to sequential
+    * streams.
+    */
+  def stepper[S <: Stepper[_]^{this}](implicit shape: StepperShape[A, S]): S = {
+    import convert.impl._
+    val s: Any = shape.shape match {
+      case StepperShape.IntShape    => new IntIteratorStepper   (iterator.asInstanceOf[Iterator[Int]])
+      case StepperShape.LongShape   => new LongIteratorStepper  (iterator.asInstanceOf[Iterator[Long]])
+      case StepperShape.DoubleShape => new DoubleIteratorStepper(iterator.asInstanceOf[Iterator[Double]])
+      case _                        => shape.seqUnbox(new AnyIteratorStepper[A](iterator))
+    }
+    s.asInstanceOf[S]
+  }
+
+  /** @return The number of elements in this $coll, if it can be cheaply computed,
+   *  -1 otherwise. Cheaply usually means: Not requiring a collection traversal.
+   */
+  def knownSize: Int = -1
+}
+
+final class IterableOnceExtensionMethods[A](private val it: IterableOnce[A]) extends AnyVal {
```

Comment:
Should `it` be impure? otherwise the extension methods only apply on pure `IterableOnce`s.

Reply:
I left it like that since all extension methods are deprecated anyway,